compiler/qsc_project/src/project.rs

```diff
@@ -61,34 +60,36 @@ pub trait FileSystemAsync {
     /// Given a path, list its directory contents (if any).
     /// This function should only return files that end in *.qs and folders.
     async fn list_directory(&self, path: &Path) -> miette::Result<Vec<Self::Entry>>;
-
-    /// Given an initial path and some regexes to exclude, fetch files that don't match
-    /// those regexes.
-    async fn fetch_files_with_exclude_pattern(
+    /// Given an initial path, fetch files matching <initial_path>/**/*.qs
+    async fn collect_project_sources(
         &self,
-        exclude_patterns: &[Regex],
-        exclude_files: &[String],
         initial_path: &Path,
     ) -> miette::Result<Vec<Self::Entry>> {
         let listing = self.list_directory(initial_path).await?;
         let mut files = vec![];
-        for item in listing {
-            let file_name = item.entry_name();
-            let name = item.path().to_string_lossy().to_string();
-            if regex_matches(exclude_patterns, &name) || exclude_files.contains(&file_name) {
-                continue;
+        for item in filter_hidden_files(listing.into_iter()) {
+            match item.entry_type() {
+                Ok(EntryType::File) if item.entry_extension() == "qs" => files.push(item),
```

Comment:
Doesn't this line mean that .qs files that are peers to the src folder will also be included? I think this pattern should be updated to match non-async version below which only begins gathering files once it has found the src folder.